### ISSUE TITLE
Find a good technology for creating a dataset description website

### ISSUE THREAD

**Issue by yohell** [NBISweden/aida-data-hub#62]

**Comment by pontus** [comment]
I have a fairly strong preference for a static generator as that makes our options for serving much wider.

**Comment by pontus** [comment]
As for XML, given that the designated metadata formats are XML-speced it would seem natural. The two reasons I can see for why not would be if one considers it makes the risk of someone submitting non-public metadata too large.

Another possible reason would be if the flow would involve getting data from the original submission (and that for some reason being unwilling to serve as XML). This doesn't seem to be the case for now, though.

**Comment by pontus** [comment]
As for additional tooling required, I guess that will be seen when/if we try to populate a demo instance with specific datatsets we've been allowed to use (I hope we won't have to parse/analyze anything else, but I don't know).

**Comment by yohell** [comment]
Regarding this:

> The two reasons I can see for why not would be if one considers it makes the risk of someone submitting non-public metadata too large.

This risk is independent of technology choice.

However, submissions to Bigpicture is serious business, and can only be made by people authorized by their organization to carry out this task in it name. They Must do this correctly, or there will be GDPR issues. The benefit of making this part of metadata submission, and in the XML format, is that this will ensure that the attention of this person with the required authority is on this data at the point of submission, and the data would then be in the same format that this person must sign off on anyway, so this choice adds effectively no added cognitive overhead.

**Comment by yohell** [comment]
Regarding this:

> when/if we try to populate a demo instance with specific datatsets we've been allowed to use

Our workflow will probably be that eithe 1) we generate a proposal based on a script and stored data, and ask each data contributor for confirmation before putting the proposal online, or 2) we ask data contributors to just send us the data we need. 

I think 1 is feasible, and suspect 2 is not.

**Comment by pontus** [comment]
Without trying to impose my opinions or stopping anyone from looking, I think Hugo (https://gohugo.io/) matches our needs fairly well (and have some very nice extras, being Golang so it's just a single binary to deal with, no dependency issues).

**Comment by darthvader2** [comment]
Yes, Hugo seem to be a fair choice. Other frameworks like Gatsby, Next.js have a steep learning curve(One need to learn JS and React JS). On the other hand Hugo uses GO templates, which we deal with everyday for Helm charts and K8s.

**Comment by shreyasshivakumara** [comment]
I tried to play around a bit using Hugo. It seems to be supporting XML files pretty well. I vouch for it but it would be to have a checklist of things that this technology particularly needs. I also think this card can be broken down into multiple cards.

**Comment by pontus** [comment]
Try a conversion and see how it goes.
